### ISSUE TITLE
Creating an auto scaling group removes  the previously existing instances from the load balancer

### ISSUE THREAD

**Issue by gipi** [spulec/moto#3748]
I'm experiencing a weird behaviour with ``moto`` trying to unit test some fabric scripts used to manage code on machines under a load balancer: for the real setup I'm using ``ansible`` and i'm creating a couple of machines, registering them when the load balancer is created; at this point I'm also associating an autoscaling group to the same load balancer. On the real AWS, querying with ``describe_load_balancers()`` the same load balancer I obtain two sets of machines, each one associated with a "Reservation".

With ``moto`` instead the ``create_auto_scaling_group()`` removes the previous instances and I obtain only one set of machines, i.e. the machines associated with the autoscaling group.

It's not clear to me if it's really a bug of ``moto`` but since this is something that happens with the **real** services I thought was worthy a shot and someone can help me.

By the way this is the code used for ansible

```yaml
- name: Provision a set of instances for web
  ec2:
    region: eu-west-1
    key_name: ssh_xxx
    group:
      - "{{ security_group_webapp.group_name }}"
      - "{{ security_group_ssh.group_name }}"
    instance_type: t2.micro
    image: "ami-xxx"
    vpc_subnet_id: "{{ subnet.subnet.id }}"
    assign_public_ip: true
    wait: true
    exact_count: 1
    instance_tags:
      Name: xxx
  register: ec2

- name: Extract EC2 instances ids
  set_fact:
    ec2_ids: "{{ ec2_ids | default([]) |  union([item.id]) }}"
  with_items: "{{ ec2.tagged_instances }}"

- name: Setup ELB
  ec2_elb_lb:
    name: "xxx-elb"
    state: present
    region: "eu-west-1"
    subnets: "{{ subnet.subnet.id }}"
    instance_ids: "{{ ec2_ids }}"
    security_group_ids:
      - "{{ security_group_webapp.group_id }}"
    # validate_certs: no
    listeners:
      - protocol: https # options are http, https, ssl, tcp
        load_balancer_port: 443
        instance_port: 443
        ssl_certificate_id: "arn:aws:iam::027188111508:server-certificate/xxx"
    health_check:
      ping_protocol: https
      ping_port: 443
      ping_path: "/"
      response_timeout: 5 # seconds
      interval: 30 # seconds
      unhealthy_threshold: 2
      healthy_threshold: 2
    wait: yes
    tags:
      Environment: test
      Application: xxx

 - name: Create launch config
   community.aws.ec2_lc:
     name: "xxx-launch-configuration"
     image_id: "ami-xxx"
     key_name: ssh_xxx
     region: "eu-west-1"
     # security_groups:
     instance_type: t2.micro
     assign_public_ip: yes

 - name: Setup Auto Scaling Group for load balancer
   community.aws.ec2_asg:
     name: "xxx-autoscaling-group"
     load_balancers: [ 'xxx-elb',]
     availability_zones: [ 'eu-west-1a',]
     launch_config_name: "xxx-launch-configuration"
     min_size: 4
     max_size: 10
     desired_capacity: 2
     vpc_zone_identifier: [ "{{ subnet.subnet.id }}" ]
     tags:
       - Environment: test
         Application: xxx
```

Instead the python code is the following

```python
    ec2_resource = boto3.resource('ec2')
    instances = ec2_resource.create_instances(ImageId=ami_id, MinCount=3, MaxCount=5)
    # random name for the elb
    prefix = ''.join([chr(random.randint(ord('a'), ord('z'))) for _ in range(16)])
    elb_name = "%s-elb" % prefix
    # create a load balancer with such name
    elb_instance = elb_client.create_load_balancer(
        LoadBalancerName=elb_name,
        Listeners=[
            {
                'Protocol': 'string',
                'LoadBalancerPort': 123,
                'InstanceProtocol': 'string',
                'InstancePort': 123,
                'SSLCertificateId': 'string'
            },
    ])
    instance_ids = [_.id for _ in instances]
    # attach the instances to the load balancer
    r_registering = elb_client.register_instances_with_load_balancer(
        LoadBalancerName=elb_name,
        Instances=[{'InstanceId': _} for _ in instance_ids[1:]],
    )

    asg_name = 'kebab-asg'
    lc_name = 'kebab-lc'
    response = asg_client.create_launch_configuration(
        LaunchConfigurationName=lc_name,
        ImageId=ami_id,
        InstanceType="t1.micro",
    )

    # create an autoscaling group for such ELB
    response = asg_client.create_auto_scaling_group(
        AutoScalingGroupName=asg_name,
        LoadBalancerNames=[elb_name],
        LaunchConfigurationName=lc_name,
        AvailabilityZones=['eu-west-1a'],
        MinSize=10,
        MaxSize=100,
    )

```

Looking in the code I found that the lines in ``moto/autoscaling/models.py`` are responsible for this behaviour, in particular line 946 for the class ``AutoScalingBackend``

```python
        for elb in elbs:
            elb_instace_ids = set(elb.instance_ids)
            self.elb_backend.register_instances(
                elb.name, group_instance_ids - elb_instace_ids
            )
            self.elb_backend.deregister_instances(
                elb.name, elb_instace_ids - group_instance_ids
            )
```

to me seems that it's adding the instances in the auto scaling group that are not already associated with the load balancer, but **it's removing all the instances that are not in the group**.

**Comment by bblommers** [comment]
Hi @gipi, thanks for raising this. Looks like a bug indeed - like we never expect a mix of standalone instances and autoscaling-groups.

Looking at the history, this has been the behaviour for about 5 years now, so I guess there are not a lot of people who use ELB's like this.
If you want to contribute a fix, PR's are always welcome!

**Comment by gipi** [comment]
Ok, I'll try next week to come up with something, thanks for the answer.

**Comment by gipi** [comment]
Should be fixed with #3757
